query/src/sql/planner/semantic/type_check.rs

```diff
@@ -438,10 +469,65 @@ impl<'a> TypeChecker<'a> {
         child: &Expr<'a>,
         required_type: Option<DataTypeImpl>,
     ) -> Result<(Scalar, DataTypeImpl)> {
-        self.resolve_function(op.to_string().as_str(), &[child], required_type)
+        if *op == UnaryOperator::Plus {
+            // Omit unary + operator
+            return self.resolve(child, required_type).await;
+        }
+        let op_name = match op {
+            UnaryOperator::Minus => "negate".to_string(),
+            _ => op.to_string(),
+        };
```

Comment:
Explicitly match against `Plus` and `Not`?

Reply:
Done.